src/rgw/rgw_rest_pubsub.cc

```diff
@@ -171,12 +296,35 @@ class RGWPSGetTopic_ObjStore_AWS : public RGWPSGetTopicOp {
   }
 };
 
+void RGWPSGetTopicOp::execute(optional_yield y) {
+  op_ret = get_params();
+  if (op_ret < 0) {
+    return;
+  }
+  ps.emplace(static_cast<rgw::sal::RadosStore*>(store), s->owner.get_id().tenant);
+  op_ret = ps->get_topic(topic_name, &result);
+  if (topic_has_endpoint_secret(result) && !verify_transport_security(s->cct, *(s->info.env))) {
+    ldpp_dout(this, 1) << "topic '" << topic_name << "' contain secret and cannot be sent over insecure transport" << dendl;
+    op_ret = -EPERM;
+    return;
+  }
+  if (op_ret < 0) {
```

Comment:
op_ret < 0 here means get_topic() returned it, right?
so why didn't we react to that after line 305?